nexusutils/dataconverter/readers/xps/reader_utils.py

```diff
@@ -0,0 +1,835 @@
+                 root_element: EmtT.Element,
+                 vendor_name: str = "specs") -> None:
+        """Collect mandatory inputs.
+
+        Parameters
+        ----------
+        root_element : Root element xml XPS data file
+        vendor_name :  Name of vendor of XPS machine
+        """
+
+        self._root_element = root_element
+
+        self._root_path = f'/ENTRY[entry]/{vendor_name}'
+        self._xps_dict: dict = {}
+        self.entry_to_data: dict = {}
+        # self.tail_part_frm_seq = ""
+        self.tail_part_frm_struct = ""
+        self.tail_part_frm_othr = ""
+        self.child_nm_reslvers = "__child_name_resolver__"
+
+    def parse_xml(self):
+        """Start parsing process and parse children recursively.
+
+        Parameters
+        ----------
+        """
+        element = self._root_element
+        element.attrib[self.child_nm_reslvers] = []
+        child_num = len(element)
+        parent_path = self._root_path
+        skip_child = -1
+
+        child_elmt_ind = 0
+        while child_num > 0:
+            self.pass_child_through_parsers(element,
+                                            parent_path,
+                                            child_elmt_ind,
+                                            skip_child)
+
+            child_num -= 1
+            child_elmt_ind += 1
+
+        self.collect_raw_data_to_construct_data()
+        self.construct_data()
+
+    def pass_child_through_parsers(self,
+                                   element_: EmtT.Element,
+                                   parent_path: str,
+                                   child_elmt_ind: int,
+                                   skip_child: int) -> None:
+        """
+        Parse the element to parser according to element tag.
+        Parameters
+        ----------
+        element_ : xml element to parse
+        parent_path : Xpath of the parent element where the element_ belongs
+        child_elmt_ind : Index of child element to track the children.
+        skip_child : Tack the children who will be skipped to pass to
+                     the parser
+        Returns
+        -------
+        None
+        """
+
+        name_val_elmt_tag = ['ulong',
+                             'double',
+                             'string',
+                             'boolean',
+                             'enum',
+                             'any']
+
+        parent_element = element_
+        element = parent_element[child_elmt_ind]
+        element.attrib['__parent__'] = parent_element  # type: ignore[assignment]
+        element.attrib['__odr_siblings__'] = child_elmt_ind  # type: ignore[assignment]
+
+        if self.child_nm_reslvers not in parent_element.attrib.keys():
+            parent_element.attrib[self.child_nm_reslvers] = []  # type: ignore[assignment]
+
+        elmt_tag = element.tag
+
+        if child_elmt_ind <= skip_child:
+            pass
+
+        elif elmt_tag == "sequence":
+
```

Comment:
Why the empty space? Also in between these if-elif-blocks we have too many spaces. Not really needed are they?